nl/nl_linux.go

```diff
@@ -585,30 +585,33 @@ func SubscribeAt(newNs, curNs netns.NsHandle, protocol int, groups ...uint) (*Ne
 }
 
 func (s *NetlinkSocket) Close() {
-	syscall.Close(s.fd)
-	s.fd = -1
+	fd := int(atomic.LoadInt32(&s.fd))
```

Comment:
If you want to avoid the race with the GetFd without using a mutex then you should use: `fd := SwapInt32(&s.fd, -1)`